app/(white_navbar)/explore/RetroPGF4/_component/Tab/component/InputAndFilterBadgeholderTab.tsx

```diff
@@ -0,0 +1,211 @@
+import { Dispatch, FC, SetStateAction } from 'react'
+import {
+  CheckBoxStateBadgeholderType,
+  CheckBoxStateType,
+  ExploreRoundState,
+} from '../../ExploreRoundType'
+import Input from '@/app/component/Input/Input'
+import { badgeholderStatus, category, multiplyOpenSource } from '../../Text'
+import Search from '@carbon/icons-react/lib/Search'
+import Grid from '@carbon/icons-react/lib/Grid'
+import List from '@carbon/icons-react/lib/List'
+import SettingsAdjust from '@carbon/icons-react/lib/SettingsAdjust'
+import SortExploreRound from '../../SortExploreRound'
+import SortBadgeholder from './SortBadgeholder'
+
+interface IInputAndFilterBadgeholderTabProps {
+  search: string
+  setCheckBox: Dispatch<SetStateAction<any>>
+  setSearch: Dispatch<SetStateAction<string>>
+  setState: Dispatch<SetStateAction<any>>
+  checkBox: CheckBoxStateBadgeholderType
+  state: ExploreRoundState
+  handleClearFilter(): void
+  handleChangeStatus(char: string): void
+  handleChangeMultiplyOpenSource(char: string): void
+  handleChangeSort(char: string): void
+}
+
+const InputAndFilterBadgeholderTab: FC<IInputAndFilterBadgeholderTabProps> = ({
+  search,
+  setCheckBox,
+  setSearch,
+  setState,
+  checkBox,
+  state,
+  handleClearFilter,
+  handleChangeStatus,
+  handleChangeMultiplyOpenSource,
+  handleChangeSort,
+}) => {
+  function handleSearchChange(e: React.ChangeEvent<HTMLInputElement>) {
+    setSearch((prev) => e.target.value)
+  }
+
+  return (
+    <>
+      <div className="flex flex-col gap-4 ">
+        <div className="flex flex-wrap items-center gap-3 lg:gap-4 mt-4">
+          <div className="flex-grow">
+            <Input
+              Icon={<Search size={24} className="fill-gray-500" />}
+              value={search}
+              onChange={handleSearchChange}
+              className="
+                            w-full pl-12  border placeholder-slate-500 bg-white text-slate-800 focus:ring focus:ring-primaryRed px-1 py-1.5 rounded-full  min-h-6 lg:min-h-10
+                            "
+              placeholder="Search Badgeholder Name"
+            />
+          </div>
+          <div className="">
+            {/* Open drawer btn */}
+            <div
+              onClick={() => {
+                setState((prev: { drawer: any }) => ({
```

Comment:
should be boolean {drawer: boolean}
But if already give setState a correct type I think no need to clarify a type again